File: CsharpProjects/myLeetCode/interview/amd/countSetBits.cs
// Write an efficient program to count the number of 1s in the binary representation of an integer.
#include <stdio.h>

/* Function to get no of set bits in binary
   representation of positive integer n */
unsigned int countSetBits(unsigned int n)
{
    unsigned int count = 0;
    while (n) {
        count += n & 1;
        n >>= 1;
    }
    return count;
}

/* Program to test function countSetBits */
int main()
{
    int i = 5;
    printf("The number of set bits in %d is: %d\n", i, countSetBits(i));
    return 0;
}

File: C++/代码随想录/interview/countSetBits.py
def countBits(n):
    count = 0
    while n:
        n &= n - 1  # 消除最右边的1
        count += 1
    return count

# 测试用例
print(countBits(5))  # 二进制101，输出: 2
print(countBits(7))  # 二进制111，输出: 3
print(countBits(0))  # 二进制0，输出: 0
